### ISSUE TITLE
Upgrade Webapp to use Bootstrap 4 or 5

### ISSUE THREAD

**Issue by dianabarsan** [medic/cht-core#8176]
**What feature do you want to improve?**
We use a very old version of `ngx-bootstrap@^7`, which can no longer be installed without passing `legacy-peer-deps`, because it lists an older Angular version as a peer dependency - it supports up to Angular 12.

At the same time, we use [Bootstrap 3 in webapp](https://github.com/medic/cht-core/blob/a30c1649e129cca54a42cc4e60976ce73a314af9/webapp/src/ts/app.component.ts#L146).  
[Support for `bootstrap3` was dropped in `ngx-bootstrap@10`](https://github.com/valor-software/ngx-bootstrap/blob/development/CHANGELOG.md#breaking-changes). 
The latest version of `ngx-bootstrap` still supports bs3 is `9.0.0`, which supports up to Angular 14.  

**Describe the improvement you'd like**
Update webapp to use a newer version of Bootstrap. Bootstrap 4 might be the easiest migration, with minimal changes required. 

**Describe alternatives you've considered**
- Drop `ngx-bootstrap`?
- We're fine with installing with `legacy-peer-deps` and ignore compatibility mismatches

**Additional context**
I wanted to remove the need for installing with `legacy-peer-deps`, this was the only true blocker.

**Comment by dianabarsan** [comment]
Can we completely eliminate Bootstrap with Angular material?

**Comment by garethbowen** [comment]
> Can we completely eliminate Bootstrap with Angular material?

No. Material replaces the bootstrap widgets but not a lot of the layout we use. Care Teams is doing a great job of replacing the widgets.

Eventually we want to drop bootstrap as it no longer works with our design but it's a big effort high risk change. Any steps we can take in this direction would be good. I think dropping ngx-bootstrap would be a great first step.

I think there are two options to remove ngx-bootstrap...

1. Write custom glue that wires bootstrap to angular. This is what we used to do before we started using ngx-bootstrap so we didn't have to do it ourselves. Probably not too difficult but also feels like a step backwards.
2. It looks like we're using it for modals, dropdowns, and accordions. These could be replaced with Material components [Dialog](https://material.angular.io/components/dialog/overview), [Select](https://material.angular.io/components/select/overview), and [Expansion Panel](https://material.angular.io/components/expansion/overview) (respectively). We'd have to be a little careful that the new design doesn't look too weird with the old design (eg: a Material Select beside a bootstrap Input is probably a bad idea). This could potentially be fixed in CSS but it probably makes more sense to use option 1 for dropdowns, and option 2 for modals and accordions.
